extension/goplugin/integration_tests/end_to_end_test.go

```diff
@@ -218,6 +218,7 @@ var _ = Describe("Environment", func() {
 			Expect(result).To(HaveKeyWithValue("error", "Validation error: Json validation error:\n\tname: String length must be greater than or equal to 3,"))
 			resource["name"] = "abcd"
 			expectedResponse["name"] = "abcd"
+			delete(expectedResponse, "subobject")
```

Comment:
Why this is not returned by PUT? We know from BeforeEach that resource has this field and I don't see anything being removed.

Reply:
actually, this was something which also confused me and i thought it was a bug but it seems is expected behavior. We do not return nested objects which are `nil` only when `*_in_transaction` handler is defined (previously there was only `pre_create` handler that's why `subobject` was returned)

Reply:
Ack. Seems inconsistent and problematic but this is not a problem with this PR, approving.